### PR TITLE
Fix: storage patch env

### PR DESCRIPTION
**Pull request by Somefive**
### Description of your changes

Fixes storage patch env bug in #3300

P.S. This PR does not ensure all the possible cases in storage works. It only fixes the bug in #3300.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by barnettZQG — APPROVED** [review]
LGTM

File: pkg/cue/model/sets/operation.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func listMergeProcess(field *ast.Field, key string, baseList, patchList *ast.Lis
 			kmaps[fmt.Sprintf(key, blit.Value)] = patchList.Elts[i]
 		}
 		if !foundPatch {
-			return
+			continue
```
Comment (FogDong) [inline pkg/cue/model/sets/operation.go:118]: Can you add a test in application_controller?

Comment (Somefive) [reply pkg/cue/model/sets/operation.go:118]: An e2e test is added.

Comment (Somefive) [reply pkg/cue/model/sets/operation.go]: UT is also added.

**Review by FogDong — APPROVED** [review]

**Pull request merged**